pages/establishment/update/index.js

```diff
@@ -0,0 +1,80 @@
+const { omit } = require('lodash');
+const { page } = require('@asl/service/ui');
+const { form } = require('../../common/routers');
+const schema = require('./schema');
+const confirm = require('./routers/confirm');
+const success = require('../../common/routers/success');
+const { groupFlags, ungroupFlags } = require('../../establishment/formatters/flag-grouping');
+
+module.exports = settings => {
+  const app = page({
+    root: __dirname,
+    paths: ['/confirm', '/success'],
+    ...settings
+  });
+
+  app.use((req, res, next) => {
+    req.breadcrumb('establishment.update');
+    req.model = groupFlags(req.establishment);
+
+    next();
+  });
+
+  app.use(form({
+    schema: Object.assign({}, schema, {
+      comments: {
+        inputType: 'textarea',
+        validate: ['required']
+      }
+    }),
+    process: (req, res, next) => {
+      if (!Array.isArray(req.body.licences)) {
+        req.form.values.licences = req.body.licences ? [req.body.licences] : [];
+      }
+      if (!Array.isArray(req.body.authorisations)) {
+        req.form.values.authorisations = req.body.authorisations ? [req.body.authorisations] : [];
+      }
```

Comment:
I'm not sure I follow what happens here if it _is_ an array.

Reply:
I think this was an issue with the POSTed form values coming through as a string if a single checkbox was selected, but as an array if multiple were selected.

Reply:
I get that, this just doesn't seem to handle the case where it _is_ an array.

Reply:
That's already handled by the form router IIRC - this just overwrites the saved form values in the cases where the form returns a string or null. I think these cases are usually handled by the form router but because these are pseudo-fields and not actual properties of the establishment model, it doesn't seem to work without this code.